src/hotspot/cpu/x86/x86.ad

```diff
@@ -8636,15 +8636,18 @@ instruct loadShuffleI_avx(vec dst, vec src) %{
 
 instruct rearrangeI_avx(vec dst, vec src, vec shuffle) %{
   predicate((Matcher::vector_element_basic_type(n) == T_INT || Matcher::vector_element_basic_type(n) == T_FLOAT) &&
-            UseAVX >= 2);
+            UseAVX > 0);
   match(Set dst (VectorRearrange src shuffle));
   format %{ "vector_rearrange $dst, $shuffle, $src" %}
   ins_encode %{
     int vlen_enc = vector_length_encoding(this);
     if (vlen_enc == Assembler::AVX_128bit) {
-      vlen_enc = Assembler::AVX_256bit;
+      __ vpermilps($dst$$XMMRegister, $src$$XMMRegister, $shuffle$$XMMRegister, vlen_enc);
+    } else if (Matcher::vector_element_basic_type(this) == T_INT) {
+      __ vpermd($dst$$XMMRegister, $shuffle$$XMMRegister, $src$$XMMRegister, vlen_enc);
+    } else {
+      __ vpermps($dst$$XMMRegister, $shuffle$$XMMRegister, $src$$XMMRegister, vlen_enc);
```

Comment:
Please move this to a macro assembly routine.

Reply:
Done